src/xwayland.rs

```diff
@@ -15,6 +15,14 @@ use crate::{
     x11::{self, get_window_name},
 };
 
+pub trait Serialize {
```

Comment:
Instead of using a custom trait here, we could just implement the `Display` trait and the `TryFrom<&str>` traits.